library/src/main/java/com/google/android/exoplayer/util/BitArray.java

```diff
@@ -151,7 +151,7 @@ public int readUnsignedByte() {
     byte b;
```

Comment:
By the way, the problem isn't that the shift is signed. It's that data[byteOffset+1] is cast from byte to int before the shift happens. If the byte is negative, the cast does more than just adding zeros to the front.

Your fix is correct, but I feel the method could be written more clearly (and without an unnecessary int->byte->int conversion), like:

```
  public int readUnsignedByte() {
    int value;
    if (bitOffset != 0) {
      value = ((data[byteOffset] & 0xFF) << bitOffset)
          | ((data[byteOffset + 1] & 0xFF) >>> (8 - bitOffset));
    } else {
      value = data[byteOffset] & 0xFF;
    }
    byteOffset++;
    return value;
  }
```

Thoughts? Happy for you to amend your change to do this, or I can submit it separately.